### DISCUSSION TITLE
Get-d365lcsapitoken and passwords with special characters like $

### DISCUSSION THREAD

**Discussion by gammelgaard52** [d365collaborative/d365fo.tools#542] [Q&A]
Hi,

Does anybody have some input on how to handle passwords with special characters like $ in a DevOps release pipeline?
When I use a DevOps variable, marked as secret, with password: 123$abc - the API returns errors because of faulty password.

I can make it work from my local Visual Studio code, by using escape characters or using single quotes around hard-coded password. But when I use the DevOps variable like $(LcsPassword), it won't work. I've also tried to implement logic in the DevOps pipeline, like this:

$pw = $(LcsPassword)
$pw = $pw -replace ([Regex]::Escape("$")),"`$"

and 

$pw = $(LcsPassword)
$pw = $pw -replace "$","`$"

Any ideas?

**Comment by celuxcwb** [comment]
Usually in these cases I put my password in single quotes,

example:

'56Fg9u8$%"0ca'

it has worked for me

**Reply by udaymuppana** [reply]
Please put the password in single quotes. It will work. 
We had the issue when initially kept in double quotes "", especially if you have $ symbol as part of your password.
